tests/nnstreamer_filter_python3/unittest_filter_python3.cc

```diff
@@ -0,0 +1,323 @@
+}
+
+/**
+ * @brief Test subplugin existence.
+ */
+TEST (nnstreamerFilterPython3, checkExistence)
+{
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("python3");
+  EXPECT_NE (sp, nullptr);
+}
+
+/**
+ * @brief Negative test case with invalid model file path
+ */
+TEST (nnstreamerFilterPython3, openClose00_n)
+{
+  int ret;
+  void *data = NULL;
+  const gchar *model_files[] = {
+    "some/invalid/model/path.py",
+    NULL,
+  };
+  GstTensorFilterProperties prop;
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("python3");
+  EXPECT_NE (sp, nullptr);
+
+  _SetFilterProp (&prop, "python3", model_files);
+  ret = sp->open (&prop, &data);
+  EXPECT_NE (ret, 0);
+}
+
+/**
+ * @brief Positive case with open/close
+ */
+TEST (nnstreamerFilterPython3, openClose01)
+{
+  int ret;
+  void *data = NULL;
+  gchar *model_file;
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  GstTensorFilterProperties prop;
+  model_file =
+      g_build_filename (root_path, "tests", "test_models", "models",
+      "passthrough.py", NULL);
+  ASSERT_TRUE (g_file_test (model_file, G_FILE_TEST_EXISTS));
+
+  const gchar *model_files[] = {
+    model_file,
+    NULL,
+  };
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("python3");
+  EXPECT_NE (sp, nullptr);
+  _SetFilterProp (&prop, "python3", model_files);
+
+  /* close before open */
+  sp->close (&prop, &data);
+
+  ret = sp->open (&prop, &data);
+  EXPECT_EQ (ret, 0);
+  sp->close (&prop, &data);
+
+  /* double close */
+  sp->close (&prop, &data);
+  g_free (model_file);
+}
+
+/**
+ * @brief Positive case with successful getModelInfo
+ */
+TEST (nnstreamerFilterPython3, getModelInfo00)
+{
+  int ret;
+  void *data = NULL;
+  gchar *model_file;
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  GstTensorFilterProperties prop;
+  model_file =
+      g_build_filename (root_path, "tests", "test_models", "models",
+      "passthrough.py", NULL);
+  ASSERT_TRUE (g_file_test (model_file, G_FILE_TEST_EXISTS));
+
+  const gchar *model_files[] = {
+    model_file,
+    NULL,
+  };
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("python3");
+  EXPECT_NE (sp, nullptr);
+  _SetFilterProp (&prop, "python3", model_files);
+
+  ret = sp->open (&prop, &data);
+  EXPECT_EQ (ret, 0);
+
+  GstTensorsInfo in_info, out_info;
+
+  ret = sp->getModelInfo (NULL, NULL, data, GET_IN_OUT_INFO, &in_info, &out_info);
+  EXPECT_EQ (ret, 0);
+
+  EXPECT_EQ (in_info.num_tensors, 1U);
+  EXPECT_EQ (in_info.info[0].dimension[0], 3U);
+  EXPECT_EQ (in_info.info[0].dimension[1], 280U);
+  EXPECT_EQ (in_info.info[0].dimension[2], 40U);
+  EXPECT_EQ (in_info.info[0].type, _NNS_UINT8);
+  EXPECT_EQ (out_info.num_tensors, 1U);
+  EXPECT_EQ (out_info.info[0].dimension[0], 3U);
+  EXPECT_EQ (out_info.info[0].dimension[1], 280U);
+  EXPECT_EQ (out_info.info[0].dimension[2], 40U);
+  EXPECT_EQ (out_info.info[0].type, _NNS_UINT8);
```

Comment:
What is the meaning of 3, 280, 40? 
I think it would be good to add some comments here or use variables such as width, height ...

Reply:
> What is the meaning of 3, 280, 40? I think it would be good to add some comments here or use variables such as width, height ...

Or you can use ```constexpr```.
For example,
```suggestion
  constexpr unit32_t WIDTH = 280;
  constexpr unit32_t HEIGHT = 40;
  
  EXPECT_EQ (in_info.num_tensors, 1U);
  EXPECT_EQ (in_info.info[0].dimension[0], 3U);
  EXPECT_EQ (in_info.info[0].dimension[1], 280U);
  EXPECT_EQ (in_info.info[0].dimension[2], 40U);
  EXPECT_EQ (in_info.info[0].type, _NNS_UINT8);
  EXPECT_EQ (out_info.num_tensors, 1U);
  EXPECT_EQ (out_info.info[0].dimension[0], 3U);
  EXPECT_EQ (out_info.info[0].dimension[1], 280U);
  EXPECT_EQ (out_info.info[0].dimension[2], 40U);
  EXPECT_EQ (out_info.info[0].type, _NNS_UINT8);
```

Reply:
Changed it to use constexpr!
Thank you :)